### ISSUE TITLE
iOS 16 Beta Home App only shows the same name for each switch 

### ISSUE THREAD

**Issue by Crash-Evans** [Jason-Morcos/homebridge-magic-occupancy#31]
I am running iOS beta 16 on all my devices and I completed all the names for my switches in the Homebridge plugin. However, once in the default Home app and Home+ app the switches all show the same name ie, kitchen occupancy. There is no way to differentiate between them all.

**Comment by Jason-Morcos** [comment]
Can you please provide screenshots? I'm running iOS 16 on one of my devices and haven't had any issues with that, but I'd also caution you in general that plugins like this one likely won't support iOS 16 until later this fall.

**Comment by Crash-Evans** [comment]
Good morning Jason,
Thank you for getting back to me so quickly. I totally get that this won’t be fully supported until this fall. That said, please see below the screenshots requested. 

I have noticed a few plugins that used to have several input types (for example Homebridge weather). Has reduced down to one input type due to the layout change. 

The fact that it works for you suggest that there is an issue on my end! I’m not seeing any issues in the logs and the plug-in spins up ok. 

Thank you in advance. 

Sam
![9981869D-4BA0-4E35-ADAE-88137596E482](https://user-images.githubusercontent.com/108951455/178100937-2e19dca9-3cea-4f3b-995f-cdc3a45dd0c5.png)
![CBFC983D-C15A-48ED-A812-8E1785B49D89](https://user-images.githubusercontent.com/108951455/178100940-04cfa674-a9db-4eae-aaa8-3fbf1c8faa37.png)
![1CCCD68F-FDFA-40E0-BB04-967ECD21F3F2](https://user-images.githubusercontent.com/108951455/178100941-07f47a68-672f-4a82-83f9-ec799b2f0017.png)
![6AF96D21-F98F-4CC6-87C2-372A3F4AB171](https://user-images.githubusercontent.com/108951455/178100943-7ce76050-9b6d-4a3b-aafa-0e00be16b4fe.png)

**Comment by Crash-Evans** [comment]
just to follow up. I’ve just set up the automations with an older device (running IOS 15), and it shows the expected titles correctly.

**Comment by adec** [comment]
> The fact that it works for you suggest that there is an issue on my end! I’m not seeing any issues in the logs and the plug-in spins up ok.

Hi Sam, it's not just you - I've got the same issue.  Suspect it's a bug in iOS 16 beta.  Now to wrestle a device from a family member to make changes to the automations.  Regards, Adrian.

**Comment by Jason-Morcos** [comment]
I haven't had a chance to look into this, but please make sure you report this to Apple via the Feedback app!

**Comment by Jason-Morcos** [comment]
As an update: I've updated my iPhone to iOS 16 Beta and am seeing the same think you folks are! This really does seem like an iOS 16 bug. Does anyone know of any other plugins with multiple switches on the same device that are _not_ affected by this same bug? Has anyone tried displaying the accessories as separate tiles? Does that fix the names?

**Comment by Crash-Evans** [comment]
Afternoon Jason,
I have tried I grouping the devices and I see the same behaviour. As mentioned above, I have sent feedback. 

It’s odd, the behaviour of the switches is exactly how it should be. It’s just the visual element that doesn’t seem to work. 

As an aside, I see similar behaviour in the Weather station plug-in. I’m iOS 16, it has removed some of the sensors that used to be occupancy sensors.

**Comment by nchieng** [comment]
Just adding my report here. on iOS 16 and seeing the same thing.
It's hindering me from setting up a new automation as I can't tell which switch is which.

**Comment by Jason-Morcos** [comment]
Unfortunately I really have no idea how to address this because really these are all apple-home-app-side changes. 
PRs are super welcome if folks have ideas!

I do have a few workarounds ideas though:
1. Try this idea from the Homebridge community: https://github.com/merdok/homebridge-xiaomi-fan/issues/123
2. Setup switches one at a time (add one switch in the plugin config, restart homebirdge, see just that one new one in Home app) and rename that one in the Home app at the time when you know which it is


Related: https://github.com/homebridge/homebridge/issues/3210

**Comment by Jason-Morcos** [comment]
Additionally, I encourage everyone to continue reporting this bug to Apple. It appears they don't have an interest in fixing this :/

**Comment by leandromagro** [comment]
Hi guys,
Did you find any work around? Even with the newest version, 16.1.2, nothing was fixed. 
BTW using the tips above I was able to identify the Light Switch, stateless, etc. I’ve recreated all the rules expecting to be something with the old one but still. It’s been difficult to trust the homekit reliability these days…

**Comment by roboterm** [comment]
Good evening,

here’s a workaround for you:  
1. Edit the name of the button.  
2. Erase it.  
3. The ‘real’ name is now visible and you can change it back to its prior name.  

Have a good night 🌙

**Comment by roycepope** [comment]
@roboterm Thank You! Great workaround

**Comment by ermitovski** [comment]
Hi @Jason-Morcos 
If it helps, in the issue you mentioned before, looks like they found a solution adding this to the accessory: 
```
  service.addOptionalCharacteristic(Characteristic.ConfiguredName);
  service.setCharacteristic(Characteristic.ConfiguredName, accessoryName);
```
https://github.com/homebridge/homebridge/issues/3210#issuecomment-1619419967

**Comment by Jason-Morcos** [comment]
Thank you for the tip, @ermitovski! I just released an updated version of the plugin (v3.3.0) that I think should fix this! Thank you

**Comment by ermitovski** [comment]
Thanks a lot @Jason-Morcos!! I've updated to the new version (v3.3.0) and switches now have the expected name.
